Web Greeter v4

JezerM opened the pull request:
# Major changes

- Migrate from PyQt5 to PySide6 (still in progress)
- Migrate from pip and system dependencies to uv

# TODO

- [x] Migrate to PySide6
	- [x] Replace PyQt5 libraries with PySide6
	- [x] Replace previous bridge with PySide6 implementation
- [x] Refactor whole system
	- [x] Replace current build and packaging system with uv and/or PySide6 tools
	- [x] Use new build system in GitHub Actions
	- [x] Refactor and organize the code
	- [x] Use TOML instead of YAML for config

JezerM committed:
Initial migration to pyside6

JezerM committed:
Fix initial execution

JezerM commented:
Some updates:

I'm still trying to get myself a better laptop, because working on that old HP laptop is a hell. I've been having some issues with the shipping and that kind of stuff, hopefully I can get it to be shipped on this week.

Luckily, the latest pre-release version of UTM seems to work fine with the new OpenGL and Vulkan versions that QT6 uses, so I can finally try it myself on my mac.

tuurep commented:
Thank you! I've been meaning to give this PR a proper try, although not sure if it's yet in a state I could build it, and had too many other things to do

But definitely let me know when/if there's a good time to test and I'll prioritize it :smile:

JezerM committed:
Testing some Bridge related stuff

JezerM commented:
Not yet. Even though you can already execute it, I'm still missing the Bridge stuff, the part that makes communication work between LightDM and the browser.

JezerM committed:
This seems to work...

JezerM committed:
Migrate Bridge objects to QT6

JezerM commented:
There's still some missing stuff, but the bridge now works. So, properties can be accessed and methods can be executed.

In theory, now it should be usable. I still need to test a lot of stuff first and make some improvements to the codebase before this feature is finished.

https://github.com/user-attachments/assets/ae87e5f0-52cb-430f-92ec-0c1e5738be8f

JezerM committed:
Format code with Ruff

tuurep commented:
Great, I can see my theme coming up here as well:

https://github.com/user-attachments/assets/0ec494bd-f63b-41a4-9bd7-4b123aaa5302

JezerM committed:
Migrate from yaml based config to toml

JezerM committed:
Some changes

JezerM committed:
Decouple web_page and web_logger from error_prompt

JezerM committed:
Format files

JezerM commented:
Some updates:

- Migrate from yaml based config to toml
- Some code refactor

Also, sorry for the slow updates, I've been busy with work and college.

tuurep commented:
Nice, yeah I'm definitely not in a hurry, just glad it's still proceeding :)

JezerM committed:
Fix QTimer singleShot call

JezerM committed:
Setup meson + ninja build system

JezerM committed:
Load js resources individually

JezerM committed:
Add setup for other data files

JezerM committed:
Update version to 4.0.0

JezerM committed:
Revert backlight value from delta

JezerM committed:
Use install_subdir instead of install_script

JezerM committed:
Remove makefile

JezerM committed:
Update README

JezerM committed:
Update NEWS.md

JezerM commented:
Another update:

- Use meson + ninja to build the app
- Use Nuitka with QT plugins to build a standalone app
- Fixed some small bugs

I'm only missing the GitHub actions for building the debian and ubuntu releases and it should be done.

JezerM committed:
Use proper paths

JezerM committed:
Update github actions

JezerM committed:
Fix DEBIAN files

JezerM committed:
Add arm64 build

JezerM committed:
Fix DEBIAN/postinst

JezerM commented:
Some updates:

- Fixed some build issues
- Updated GitHub Actions to use the new build system
- Added ARM builds for Debian and Ubuntu

So, anyone can try this version by either installing the Debian/Ubuntu builds or building it yourself. I'll probably merge this later today.

tuurep commented:
Thanks so much,

I'm interested in trying to build and set it up on Arch Linux, but don't have much time right now

If you want to merge before I get to that, that's good

JezerM committed:
Update README with sysconfdir override

JezerM commented:
My Framework 13 laptop just arrived yesterday, so now I'm gonna test these changes in real hardware in the following days.

I got it working easily, so it's a great start.

![image](https://github.com/user-attachments/assets/4e7c8028-9fad-4b3f-8938-3715212f0172)

tuurep commented:
Nice, congrats on the machine!

I haven't had time to look at this and also almost had to change laptops due to breaking charging ports, but still using the old one as it takes charge with the cord in a certain position.

I'm not quite sure how to build this again to test it.

JezerM commented:
You can build it with:
```
git clone --recursive https://github.com/JezerM/web-greeter.git
cd web-greeter
meson build -Dprefix=/usr -Dsysconfdir=/etc
ninja -C build
sudo ninja -C build install
```

JezerM committed:
Update themes submodule

JezerM committed:
Update DEBIAN/control version

JezerM commented:
Tested with VoidLinux:
- Works as expected
- Cannot start Wayland sessions, but I assume this is a Void and DBUS related problem

Tested with EndeavourOS in a VM:
- Works as expected
- AUR package updated locally, just waiting for the new version to be released

TODO: Test with Ubuntu and Fedora

JezerM committed:
Update README
